### ISSUE TITLE
Passing in V8 flags to a node process that executes ng build/serve throws

### ISSUE THREAD

**Issue by jackofdiamond5** [angular/angular-cli#28901]
### Command

build

### Is this a regression?

- [x] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

18.2.12

### Description

I am firing a `nodejs` process that executes the `ng build` schematic. 
For example, it looks something like this - `node --max-old-space-size=4096 node_modules/@angular/cli/bin/ng build`. This throws because of the `--max-old-space-size` V8 flag that I pass in to `node`.

I traced the problem to this [PR#28758](https://github.com/angular/angular-cli/pull/28758/files#diff-744ed3f38c651cf22c6347667c25646ae515662169495b75438a262757e3b476R63) which uses `process.execArgv` and passes them in as `execArgv` to the `options` of the newly-created worker. Later on `piscina` will spawn a worker with the provided options, without doing any filtering, but I figured this is the intended behavior on their part. 

According to `nodejs` when spawning worker threads, we cannot pass in any V8 flags ([ref](https://nodejs.org/api/worker_threads.html#new-workerfilename-options)). Also, there's an open issue https://github.com/nodejs/node/issues/41103 to make worker threads stop throwing errors for unsupported options but until that is resolved my best guess is that if `process.execArgv` is to be used, then some sort of filtering should be applied on these options before passing them in to the worker's constructor.

### Minimal Reproduction

1, Create a new application with `@angular/cli@19`.
2. Navigate to the newly-created project's root.
3. Run `node --max-old-space-size=4096 node_modules/@angular/cli/bin/ng build`

### Exception or Error

```text
✘ [ERROR] Initiated Worker with invalid execArgv flags: --max-old-space-size=4096 [plugin angular-compiler]
 node_modules/piscina/dist/index.js:130:23:
      130 │         const worker = new node_worker_threads_1.Worker((0, node_...
```

### Your Environment

```text
Angular CLI: 19.0.0
Node: 20.17.0
Package Manager: npm 10.8.2
OS: win32 x64

Angular:
...

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1900.0 (cli-only)
@angular-devkit/core         19.0.0 (cli-only)
@angular-devkit/schematics   19.0.0 (cli-only)
@schematics/angular          19.0.0 (cli-only)
```

**Comment by clydin** [comment]
Thank you for the report. This is something the team will be investigating.

For `max-old-space-size` specifically, you may not need to use that option anymore with the Angular CLI.  Newer Node.js versions have improved their memory initialization process significantly over previous versions.

**Comment by jackofdiamond5** [comment]
> For `max-old-space-size` specifically, you may not need to use that option anymore with the Angular CLI. Newer Node.js versions have improved their memory initialization process significantly over previous versions.

@clydin Yeah you're right, I don't need the `max-old-space-size` anymore, I used it as a mere example to illustrate a V8 option that would cause an error when used like that.

**Comment by viceice** [comment]
I've this problem with `--use-openssl-ca`

**Comment by janpapenbrock** [comment]
Just ran into this and found the following workaround, maybe it helps someone.

Does not work:
`node --max_old_space_size=6000 ./node_modules/@angular/cli/bin/ng build`

Does work:
`NODE_OPTIONS="--max-old-space-size=6000" node ./node_modules/@angular/cli/bin/ng build`

(Validated via: `NODE_OPTIONS="--max-old-space-size=6000" node -e 'console.log(v8.getHeapStatistics().heap_size_limit)'`)

**Comment by alan-agius4** [comment]
Thanks for reporting this issue. Luckily, it has already been fixed in one of the recent releases. Please update to the most recent version to resolve the problem.

If the problem persists in your application after upgrading, please open a new issue, provide a simple repository reproducing the problem, and describe the difference between the expected and current behavior. You can use `ng new repro-app` to create a new project where you reproduce the problem.
